fastcrypto-cli/src/ecvrf.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use clap::Parser;
+use fastcrypto::vrf::ecvrf::{ECVRFKeyPair, ECVRFPrivateKey, ECVRFProof, ECVRFPublicKey};
+use fastcrypto::vrf::{VRFKeyPair, VRFProof};
+use rand::thread_rng;
+use std::io::{Error, ErrorKind};
+
+#[derive(Parser)]
+#[command(name = "ecvrf-cli")]
+#[command(about = "Elliptic Curve Verifiable Random Function (ECVRF) over Ristretto255 according to draft-irtf-cfrg-vrf-15.", long_about = None)]
+enum Command {
+    /// Generate a key pair for proving and verification.
+    Keygen,
+
+    /// Create an output/hash and a proof.
+    Prove(ProveArguments),
+
+    /// Verify an output/hash and a proof.
+    Verify(VerifyArguments),
+}
+
+#[derive(Parser, Clone)]
+struct ProveArguments {
+    /// The hex encoded input string. May be of arbitrary length but should ideally be at least 16 bytes.
+    #[clap(short, long)]
+    input: String,
+
+    /// A hex encoding of the secret key. Corresponds to a scalar in Ristretto255 and must be 32 bytes.
+    #[clap(short, long)]
+    secret_key: String,
+}
+
+#[derive(Parser, Clone)]
+struct VerifyArguments {
+    /// Hex-encoded Sha512 hash of the proof. Must be 64 bytes.
+    #[clap(short, long)]
+    output: String,
+
+    /// Encoding of the proof to verify. Must be 80 bytes.
+    #[clap(short, long)]
+    proof: String,
+
+    /// Hex encoding of the input string used to generate the proof. May be of arbitrary length but
+    /// should ideally be at least 16 bytes.
```

Comment:
why is that preferred?

Reply:
If the prover can predict the input or it has very low entropy, the prover can create a keypair giving a specific output. @kchalkias had a tweet about it last week. But this warning should either be omitted be more clearly written.

Reply:
I've removed it. This warning is use case specific and should be somewhere else.